src/Covenant/Test.hs

```diff
@@ -68,16 +136,18 @@ instance Arbitrary Concrete where
                   BLS12_381_MlResultT
                 ]
         | otherwise =
-            oneof
-              [ pure . BuiltinFlat $ UnitT,
-                pure . BuiltinFlat $ BoolT,
-                pure . BuiltinFlat $ IntegerT,
-                pure . BuiltinFlat $ StringT,
-                pure . BuiltinFlat $ ByteStringT,
-                pure . BuiltinFlat $ BLS12_381_G1_ElementT,
-                pure . BuiltinFlat $ BLS12_381_G2_ElementT,
-                pure . BuiltinFlat $ BLS12_381_MlResultT,
-                ThunkT . Comp0 <$> (ArgsAndResult <$> liftArbitrary (go (size `quot` 4)) <*> go (size `quot` 4))
+            frequency
+              [ (10, pure . BuiltinFlat $ UnitT),
+                (10, pure . BuiltinFlat $ BoolT),
+                (10, pure . BuiltinFlat $ IntegerT),
+                (10, pure . BuiltinFlat $ StringT),
+                (10, pure . BuiltinFlat $ ByteStringT),
+                (10, pure . BuiltinFlat $ BLS12_381_G1_ElementT),
+                (10, pure . BuiltinFlat $ BLS12_381_G2_ElementT),
+                (10, pure . BuiltinFlat $ BLS12_381_MlResultT),
+                (2, ThunkT . Comp0 <$> (ArgsAndResult <$> liftArbitrary (go (size `quot` 4)) <*> go (size `quot` 4)))
```

Comment:
Not going to question this, but it's amazing that you needed to make thunks _five times_ less likely than non-thunks to reign this in.

Reply:
Realistically it could be a bit higher for automated testing, but for repl testing/development it's quite important that things be _readable_ at a glance so as to not lose immense amounts of time trying to determine what I'm even looking at.